File: RetrofitReservation/app/src/main/java/ma/ensa/retrofitreservation/adapters/ReservationAdapter.java
package ma.ensa.retrofitreservation.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import ma.ensa.retrofitreservation.ListReservationActivity;
import ma.ensa.retrofitreservation.R;
import ma.ensa.retrofitreservation.models.Reservation;

public class ReservationAdapter extends RecyclerView.Adapter<ReservationAdapter.ViewHolder> {

    private List<Reservation> reservations;
    private ListReservationActivity activity;

    public ReservationAdapter(List<Reservation> reservations) {
        this.reservations = reservations;
        this.activity = activity;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_reservation, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        Reservation reservation = reservations.get(position);
        holder.clientName.setText(reservation.getClient().getNom() + " " + reservation.getClient().getPrenom());
        holder.roomType.setText(reservation.getChambre().getTypeChambre().toString());
        holder.roomPrice.setText(String.valueOf(reservation.getChambre().getPrix()));
    }

    @Override
    public int getItemCount() {
        return reservations.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {

        TextView clientName, roomType, roomPrice;

        public ViewHolder(View itemView) {
            super(itemView);
            clientName = itemView.findViewById(R.id.client_name);
            roomType = itemView.findViewById(R.id.room_type);
            roomPrice = itemView.findViewById(R.id.room_price);
        }
    }
}

File: ReservationRetrofit_KOTLIN/app/src/main/java/ma/ensaj/reservationretrofit_kotlin/adapters/ReservationAdapter.kt
package ma.ensaj.reservationretrofit_kotlin.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import ma.ensaj.reservationretrofit_kotlin.ListReservationActivity
import ma.ensaj.reservationretrofit_kotlin.R
import ma.ensaj.reservationretrofit_kotlin.models.Reservation
import java.lang.String
import kotlin.Int
import kotlin.plus
import kotlin.toString
class ReservationAdapter(
    private val reservations: List<Reservation>
) : RecyclerView.Adapter<ReservationAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_reservation, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val reservation = reservations[position]
        holder.clientName.text = "${reservation.client?.nom} ${reservation.client?.prenom}"
        holder.roomType.text = reservation.chambre?.typeChambre.toString()
        holder.roomPrice.text = reservation.chambre?.prix.toString()
    }

    override fun getItemCount(): Int {
        return reservations.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val clientName: TextView = itemView.findViewById(R.id.client_name)
        val roomType: TextView = itemView.findViewById(R.id.room_type)
        val roomPrice: TextView = itemView.findViewById(R.id.room_price)
    }
}
